guidelines/enhancement_template.md

```diff
@@ -461,6 +462,19 @@ Describe how to
   - Namespaces deletion will not delete all objects in etcd, leading to zombie
     objects when another namespace with the same name is created.
 
+#### Operating at scale
+
+It is of special relevance the declarative nature of APIs when operating
+features on multiple clusters. If applicable, reason about any peculiarity to
+the API that does not follow this principle and thus may require specific
+considerations when operated at scale.
```

Comment:
This section could be more direct and straightforward. Here's some suggested wording based on both of your new paragraphs.

```suggestion
Simple, declarative APIs are especially important when operating features on multiple clusters such as through ACM or in our managed services products. Consider how this API will work at scale, especially in automated systems without a human administrator.  For example, if using the API requires multiple changes to resources in a specific sequence that will increase the complexity of automation built on top of the API.

Can the API be used through a simple GitOps atomic update or does it require more complex interaction?

How does the user know when the effect of their API change is completely rolled out? Do they need to watch more than one resource?

Are the inputs based on values that must be computed uniquely for each cluster?
```

Reply:
Certainly does look better. @jewzaam what do you think about it?